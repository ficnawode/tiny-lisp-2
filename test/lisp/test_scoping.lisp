
(def x 1)
(def y 2)

; Expected: 1
(print-debug x)
; Expected: 2
(print-debug y)


(let ((x 10)) ; Shadow global x
  ; Expected: 10
  (print-debug x)
  ; Expected: 2
  (print-debug y)

  (let ((y 20) (x 100)) ; Shadow outer x and global y
    ; Expected: 120
    (print-debug (+ x y))
  )

  ; Expected: 10
  (print-debug x)
)

; Expected: 1
(print-debug x)