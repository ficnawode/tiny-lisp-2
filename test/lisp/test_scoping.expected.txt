1
2
10
2
120
10
1